### ISSUE TITLE
Lint detecting typos and bad copy-paste in getter methods

### ISSUE THREAD

**Issue by sgued** [rust-lang/rust-clippy#9769]
### What it does

Credit to `@sdroege` for [the idea](https://twitter.com/sdroege_/status/1586799257968906241)

This new lint would try to detect potential typos when writing getters for structs and unions, where the name of the method doesn't match the name of the field being accessed, and that there is another field whose name does match.

It can also detect the `_mut` versions of these functions.

### Lint Name

misnamed_getters

### Category

suspicious

### Advantage

Detects potential bugs

### Drawbacks

We can't know for sure the intent and context of the method, and there might be valid cases triggering false positives.

### Example

```rust                                                                                                                                                                                                                       
struct A {                                                                                                                                                                                                                    
    a: String,                                                                                                                                                                                                                
    b: String,                                                                                                                                                                                                                
 }                                                                                                                                                                                                                             
                                                                                                                                                                                                                              
impl A {                                                                                                                                                                                                                      
     fn a(&self) -> &str{                                                                                                                                                                                                      
        &self.b                                                                                                                                                                                                             
    }                                                                                                                                                                                                                         
}                                                                                                                                                                                                                             
```

Should probably be:

```rust
impl A {                                                                                                                                                                                                                      
     fn a(&self) -> &str{                                                                                                                                                                                                      
        &self.a                                                                                                                                                                                                             
    }                                                                                                                                                                                                                         
}   
```
